src/OSPSuite.Infrastructure.Import/Core/DataSheetCollection.cs

```diff
@@ -0,0 +1,124 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+using OSPSuite.Utility.Collections;
+
+namespace OSPSuite.Infrastructure.Import.Core
+{
+   /// <summary>
+   ///    a collection of excel dataSheet sheets
+   /// </summary>
+   public class DataSheetCollection : IEnumerable<DataSheet>
+   {
+      private readonly Cache<string, DataSheet> _dataSheets;
+
+      public DataSheetCollection()
+      {
+         _dataSheets = new Cache<string, DataSheet>(x => x.SheetName);
+      }
+
+      public void AddSheet(DataSheet sheet)
+      {
+         _dataSheets.Add(sheet);
+      }
+
+      public DataSheet GetDataSheet(string sheetName)
+      {
+         return _dataSheets[sheetName];
+      }
+
+      public DataSheetCollection GetDataSheets(IReadOnlyList<string> sheetNames)
+      {
+         var sheets = new DataSheetCollection();
+
+         foreach (var sheetName in sheetNames)
+         {
+            if (_dataSheets.Contains(sheetName))
+               sheets.AddSheet(_dataSheets[sheetName]);
```

Comment:
ok this is very weird here. A Method called GET modifies the data. Very suspicious to me

Reply:
but it doesnt. sheets that gets modified is the local variable that gets returned. we get a list of sheet names, and we add to sheets the ones we find in the collection. so the internal data does not get modified. Or am I missing something?